ingestion-beam/src/main/java/com/mozilla/telemetry/options/SinkOptions.java

```diff
@@ -147,6 +147,28 @@ static Parsed parseSinkOptions(SinkOptions options) {
    * Set all the derived fields of a {@link SinkOptions.Parsed} instance.
    */
   static void enrichSinkOptions(Parsed options) {
+    validateSinkOptions(options);
     options.setParsedWindowDuration(Time.parseDuration(options.getWindowDuration()));
   }
+
+  /** Detect invalid combinations of parameters and fail fast with helpful error messages. */
+  static void validateSinkOptions(SinkOptions options) {
+    if (options.getInputType() == InputType.pubsub && options.getOutputType() == OutputType.file
+        && options.getOutputNumShards() == 0) {
+      throw new IllegalArgumentException("Missing required parameter: "
+          + " --outputNumShards must be set to an explicit non-zero value when"
+          + " --outputType=file and the input is unbounded (--inputType=pubsub);"
+          + " Dataflow recommends starting with twice the value of --maxWorkers or 10"
+          + " (see https://github.com/apache/beam/pull/1952)");
+    }
+    if (options.getInputType() == InputType.pubsub
+        && options.getErrorOutputType() == ErrorOutputType.file
+        && options.getErrorOutputNumShards() == 0) {
+      throw new IllegalArgumentException("Missing required parameter: "
+          + " --errorOutputNumShards must be set to an explicit non-zero value when"
+          + " --errorOutputType=file and the input is unbounded (--inputType=pubsub);"
```

Comment:
There's a bunch of mostly duplicated string here but it's likely not worth the effort to dedupe it.